src/worker/image-worker.js

```diff
@@ -1,24 +1,30 @@
 importScripts('jimp.min.js');
 
-const MAX_SIZE = 1448;
-const MAX_FILE_SIZE = 310 * 1024;
-const COMPRESSION = 80;
+self.addEventListener('message', event => {
+  let MAX_SIZE = 1448;
+  let MAX_FILE_SIZE = 310 * 1024;
+  let COMPRESSION = 80;
 
-self.addEventListener('message', event =>
+  if (event.data.useProfileImageSize) {
+    MAX_SIZE = 280;
+    MAX_FILE_SIZE = 1024 * 1024;
+    COMPRESSION = 80;
+  }
 
-  Jimp.read(event.data).then((image) => {
-
-    if ((image.bitmap.width > MAX_SIZE) || (image.bitmap.height > MAX_SIZE)) {
+  Jimp.read(event.data.buffer).then(image => {
+    if (event.data.useProfileImageSize) {
+      image.cover(MAX_SIZE, MAX_SIZE);
```

Comment:
Is this width and height? If yes, does it make sense to have one value for each?

Reply:
Not sure. I just copied this part over from the other worker.